Path: Fix Drilling Op issues when using rotation feature

Russ4262 opened the pull request:
This PR addresses numerous problems that existed in the rotational use of the Drilling Op.  The issues were brought to light in the forum, [4th axis tool orientation not right in G-Code](https://forum.freecadweb.org/viewtopic.php?style=3&f=15&t=52262).

There are many changes to the rotational-related code in this PR, but they are necessary to make the existing rotational feature more usable in 0.19 since the feature is available.  A large portion of the changes pertain to simplifying the rotational integration code to align it more with the integration found in the PocketShape module.

Some of the changes are simply to synchronize (update) existing rotational methods with identical methods that initiated in the PathAreaOp module.  Eventually (in the next release after 0.19), this set of methods could be removed and replaced with references to those in the PathAreaOp module as a means to consolidated the rotational code in use throughout Path WB.

The changes in this PR do not (or, at least are not intended or known to) affect standard, non-rotational use of the Drilling op.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

Russ4262 commented:
Only the MacOS Travis CI build failed.

mlampert commented on src/Mod/Path/PathScripts/PathCircularHoleBase.py:
```diff
@@ -650,7 +603,8 @@ def visualAxis(self):
             yAx = 'yAxCyl'
-            FreeCAD.ActiveDocument.addObject("App::DocumentObjectGroup", "visualAxis")
+            # zAx = 'zAxCyl'
+            VA = FreeCAD.ActiveDocument.addObject("App::DocumentObjectGroup", "visualAxis")
```
variables should not start with an upper case, unless they are public and global.
All caps is generally thought off as a constant.

Russ4262 replied:
Fixed.

mlampert commented on src/Mod/Path/PathScripts/PathCircularHoleBase.py:
```diff
@@ -778,10 +730,11 @@ def applyRotationalAnalysis(self, obj, base, angle, axis, subCount):
+        # if obj.InverseAngle is True:
+        #    angle = -1 * angle
+        #    if math.fabs(angle) == 0.0:
+        #        angle = 0.0
```
can this be removed? is there any value in keeping the comment?

Russ4262 replied:
Done - deleted.

mlampert commented on src/Mod/Path/PathScripts/PathDrilling.py:
```diff
@@ -171,23 +172,34 @@ def circularHoleExecute(self, obj, holes):
+            if has_job:
+                obj.RetractHeight = 10
+            else:
+                obj.RetractHeight.Value = obj.StartDepth.Value + 1.0
```
I don't understand what's happening here. why is there a difference in behaviour if the job is passed in or if the job is determined?

Russ4262 replied:
If no `job` is passed in, then it is a method call due to rotation usage and needs to update the RetractHeight because the StartDepth default will be adjusted due to rotation.  Otherwise, the method is being called during op setup when `job` should be passed and is purposed to set the initial default value for RetractHeight.

Pulling out the rotational code will simplify PathWB code complexity and readability again.

Russ4262 commented:
Fixed first two of three review topics.  Third has explanation provided and the related change is a solution for fixing rotational issues identified.  I also rebased the branch on top of the fixes.
